Don't use OSC 8 links when output is not a terminal

jdholtz opened the pull request:
Fixes #2688. This uses the same method that is used by Yay to determine if color codes should be in the output. 

Since `UseColor` is also available in the `text` package, this could just use the `UseColor` variable and therefore would also be configurable and disabled when you decide to use Yay without color. I didn't add any tests for this as I didn't find it feasible to simulate a TTY that would be detected, but it'd be great to get suggestions on this to add a test for it.

jdholtz committed:
Don't use OSC 8 links when output is not a terminal

Jguer commented:
I think using `useColor` would be a good idea, especially because it lets you test both scenarios easily.

basically: `useColor=true` - all bells and whisles, `useColor=false` - programatic consumption

jdholtz committed:
Use `UseColor` for disabling OSC 8 links

jdholtz committed:
Add tests for `CreateRepoLink`

jdholtz commented:
I switched to using `UseColor` and added some tests for the `CreateRepoLink` function as well.

By the way, I think the pacman-git test fails are due to commits added in the last week. Probably [this one](https://gitlab.archlinux.org/pacman/pacman/-/commit/748e62fa58e247f22d0262517e367ce07cdc5672).

krydos commented:
Hi guys, I wonder if someone can make release off of the current master to fix the `12.5.2`. Current 12.5.2 doesn't work well in `eshell` too which apparently doesn't support those codes. 

I just pulled latest master and my eshell issue is gone. 

Just to confirm, do I understand it right, that if pacman.conf doesn't have "Color" option uncommented - this new OSC 8 links functionality isn't going to be triggered?

jdholtz commented:
> Just to confirm, do I understand it right, that if pacman.conf doesn't have "Color" option uncommented - this new OSC 8 links functionality isn't going to be triggered?

That’s correct, so if it isn’t doing that then let me know as that is a bug. You can also do it with `--color=never` which should turn off color and these links as well.
